src/include/txn_inline.h

```diff
@@ -706,6 +706,8 @@ __wt_txn_modify(WT_SESSION_IMPL *session, WT_UPDATE *upd)
     WT_ASSERT(session, !WT_IS_HS((S2BT(session))->dhandle));
 
     upd->txnid = session->txn->id;
+    upd->prepared_id = txn->prepared_id;
```

Comment:
This is not correct. When we first add the update, there is no prepared id. We need to pass it inside __wt_txn_prepare for each update. Same for prepare ts. It is set in __wt_txn_prepare.

Reply:
fixed this